src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -3,19 +3,58 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Net.Http;
     using System.Reflection;
+    using System.Threading.Tasks;
 
     internal static class HeartbeatDefaultPayload
     {
-       public static readonly string[] DefaultFields =
-        {
+        public static readonly string[] DefaultFields =
+         {
             "runtimeFramework",
-            "baseSdkTargetFramework"
+            "baseSdkTargetFramework",
+            "osVersion"
+        };
+
+        public static readonly string[] DefaultOptionalFields =
+         {
+            "processId",
+            // the  following are Azure Instance Metadata fields
+            "osType",
+            "location",
+            "name",
+            "offer",
+            "platformFaultDomain",
+            "platformUpdateDomain",
+            "publisher",
+            "sku",
+            "version",
+            "vmId",
+            "vmSize"
         };
 
-        public static void PopulateDefaultPayload(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        public static readonly string[] AllDefaultFields = DefaultFields.Union(DefaultOptionalFields).ToArray();
+
+        /// <summary>
+        /// Azure Instance Metadata Service exists on a single non-routable IP on machines configured
+        /// by the Azure Resource Manager. See <a href="https://go.microsoft.com/fwlink/?linkid=864683">to learn more.</a>
+        /// </summary>
+        private static string baseImdsUrl = $"http://169.254.169.254/metadata/instance/compute";
```

Comment:
I haven't looked at it yet. Just want to make sure you know that the same IP is used in AWS. So you'd need to expect that something will be returned to you that may be incompatible with what you'd expect from Azure

Reply:
Wow, interesting! Both [Amazon AWS VMs](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html), as well as [Google cloud VMs](https://cloud.google.com/compute/docs/storing-retrieving-metadata), have it there as well it would appear. Will add a check to ensure an Azure field is present.

Reply:
Looking into this further this morning I see that the web requests all require special setup and URI resource paths to get any data out of the service. I doubt there is any risk of getting incompatible data here aside from someone mocking the service up on their own. I don't think there is much value protecting ourselves here (aside from looking only for 200 responses/allowing for retries).